### PR TITLE
Add sample location scatter maps

### PR DESCRIPTION
**Pull request by leehart**
Re: issue #473

### CODE REVIEW

**Comment by sanjaynagi** [comment]
These are handy! tried it earlier

**Comment by leehart** [comment]
E.g.
```python
import malariagen_data
ag3 = malariagen_data.Ag3(pre=True)
drc_sample_sets = [
    '1264-VO-CD-WATSENGA-VMF00161',
    '1264-VO-CD-WATSENGA-VMF00164',
]
category_orders = {
    'admin1_name': [
            # Far North
            'Nord-Ubangi', 
            'Upper Uele',
            
            # Forest
            'Tshopo',
            
            # South West
            'Kongo-Central', 
            'Kinshasa', 
            'Mai-Ndombe', 
            'Sankuru', 

            # South East
            'Central Kasai', 
            'South Kivu',
            'Tanganyika',
            'Haut-Katanga'

        ],
}
```
```python
ag3.sample_location_mapbox(
  sample_sets=drc_sample_sets,
  marker_size=17,
  category_orders=category_orders,
  zoom=4,
)
```
Screenshot 2024-10-14 at 14 09 12

```python
ag3.sample_location_geo(
  sample_sets=drc_sample_sets,
  marker_size=13,
  opacity=0.7,
  category_orders=category_orders,
  scope='africa',
  hover_name=None,
)
````
Screenshot 2024-10-14 at 14 09 23

File: malariagen_data/anoph/sample_metadata.py [hunk 1]
```diff
@@ -1183,6 +1183,155 @@ def cohorts(
...
+    @doc(
+        summary="""
+            Plot markers on a map showing sample locations
+            as a Mapbox scatter plot.
+        """,
+        parameters=dict(
+            kwargs="Passed through to px.scatter_mapbox().",
+        ),
+    )
+    def sample_location_mapbox(
```
Comment (alimanfoo) [inline malariagen_data/anoph/sample_metadata.py:1196]:
```suggestion
    def plot_sample_location_mapbox(
```

Naming suggestion for consistency with other functions, "plot_" sets expectation the function will generate some kind of figure.

File: malariagen_data/anoph/sample_metadata.py [hunk 2]
```diff
@@ -1183,6 +1183,155 @@ def cohorts(
...
+    @doc(
+        summary="""
+            Plot markers on a map showing sample locations
+            as a geographic scatter plot.
+        """,
+        parameters=dict(
+            kwargs="Passed through to px.scatter_mapbox().",
+        ),
+    )
+    def sample_location_geo(
```
Comment (alimanfoo) [inline malariagen_data/anoph/sample_metadata.py:1270]:
```suggestion
    def plot_sample_location_geo(
```

File: tests/anoph/test_sample_metadata.py [hunk 3]
```diff
@@ -937,6 +937,36 @@ def test_plot_samples_bar(fixture, api):
     assert isinstance(fig, go.Figure)
 
 
+@parametrize_with_cases("fixture,api", cases=".")
+def test_sample_location_mapbox(fixture, api):
```
Comment (alimanfoo) [inline tests/anoph/test_sample_metadata.py:941]:
```suggestion
def test_plot_sample_location_mapbox(fixture, api):
```

File: tests/anoph/test_sample_metadata.py [hunk 4]
```diff
@@ -937,6 +937,36 @@ def test_plot_samples_bar(fixture, api):
...
+    fig = api.sample_location_mapbox(
+        sample_sets=sample_sets,
+        show=False,
+    )
+
+    assert isinstance(fig, go.Figure)
+
+
+@parametrize_with_cases("fixture,api", cases=".")
+def test_sample_location_geo(fixture, api):
```
Comment (alimanfoo) [inline tests/anoph/test_sample_metadata.py:956]:
```suggestion
def test_plot_sample_location_geo(fixture, api):
```

File: malariagen_data/anoph/sample_metadata.py [hunk 5]
```diff
@@ -1183,6 +1183,155 @@ def cohorts(
...
+        fig = px.scatter_mapbox(
+            df_locations,
+            lat="latitude",
+            lon="longitude",
+            mapbox_style="open-street-map",
+            zoom=zoom,
+            color=color,
+            category_orders=category_orders,
+            color_discrete_sequence=color_discrete_sequence,
+            hover_name=hover_name,
```
Comment (alimanfoo) [inline malariagen_data/anoph/sample_metadata.py:1245]:
```suggestion
            hover_name=hover_name,
            hover_data=location_columns,
```

Useful to get other fields to appear in hover tooltip?

File: malariagen_data/anoph/sample_metadata.py [hunk 6]
```diff
@@ -1183,6 +1183,155 @@ def cohorts(
...
+
+        fig = px.scatter_geo(
+            df_locations,
+            lat="latitude",
+            lon="longitude",
+            scope=scope,
+            height=height,
+            width=width,
+            color=color,
+            hover_name=hover_name,
```
Comment (alimanfoo) [inline malariagen_data/anoph/sample_metadata.py:1319]:
```suggestion
            hover_name=hover_name,
            hover_data=location_columns,
```

**Review by alimanfoo — COMMENTED** [review]
Thanks @leehart, looks good, couple of small suggestions.

Also don't forget to add the new functions to the Ag3 and Af1 API docs.

File: malariagen_data/anoph/sample_metadata.py [hunk 7]
```diff
@@ -1183,6 +1183,157 @@ def cohorts(
...
+        sample_query_options: Optional[base_params.sample_query_options] = None,
+        marker_size: plotly_params.marker_size = 10,
+        color: plotly_params.color = "admin1_name",
+        color_discrete_sequence: plotly_params.color_discrete_sequence = px.colors.qualitative.Prism,
+        category_orders: plotly_params.category_order = None,
+        hover_name: plotly_params.hover_name = "location",
+        fitbounds: plotly_params.fitbounds = "locations",
+        scope: plotly_params.scope = "world",
+        width: plotly_params.fig_width = 500,
+        height: plotly_params.fig_height = 500,
```
Comment (alimanfoo) [inline malariagen_data/anoph/sample_metadata.py:1285]:
```suggestion
        width: plotly_params.fig_width = 800,
        height: plotly_params.fig_height = 600,
```

Suggest to use the same default width and height as the mapbox function. 500px is a bit small.

**Review by alimanfoo — APPROVED** [review]
😎

**Pull request merged**